crates/cheatnet/src/rpc.rs

```diff
@@ -846,50 +854,76 @@ pub fn execute_constructor_entry_point(
     // endregion
 }
 
-// Inspired by blockifier::execution::syscalls::library_call
-// Calls a library using our implementation with modified logic
+// blockifier/src/execution/syscalls/mod.rs:407 (library_call)
 pub fn library_call_syscall(
     request: LibraryCallRequest,
     vm: &mut VirtualMachine,
-    syscall_handler: &mut CheatableSyscallHandler<'_>,
+    syscall_handler: &mut CheatableSyscallHandler<'_>, // Modified parameter type
     remaining_gas: &mut u64,
 ) -> SyscallResult<SingleSegmentResponse> {
+    let call_to_external = true;
+    let retdata_segment = execute_library_call(
+        syscall_handler,
+        vm,
+        request.class_hash,
+        call_to_external,
+        request.function_selector,
+        request.calldata,
+        remaining_gas,
+    )?;
+
+    Ok(SingleSegmentResponse {
+        segment: retdata_segment,
+    })
+}
+
+// blockifier/src/execution/syscalls/hint_processor.rs:577 (execute_library_call)
+pub fn execute_library_call(
+    syscall_handler: &mut CheatableSyscallHandler<'_>, // Modified parameter type
+    vm: &mut VirtualMachine,
+    class_hash: ClassHash,
+    call_to_external: bool,
+    entry_point_selector: EntryPointSelector,
+    calldata: Calldata,
+    remaining_gas: &mut u64,
+) -> SyscallResult<ReadOnlySegment> {
+    let entry_point_type = if call_to_external {
+        EntryPointType::External
+    } else {
+        EntryPointType::L1Handler
+    };
```

Comment:
I've added this method so the code mirrors the blockifier code directly, so it can be easier to update @war-in